src/json2netns/route.py

```diff
@@ -0,0 +1,103 @@
+from ipaddress import ip_address, ip_network
+from typing import Dict, List
+import logging
+from subprocess import check_output
+
+from json2netns.consts import DEFAULT_IP
+
+
+LOG = logging.getLogger(__name__)
+
+
+class Route:
+    IP = DEFAULT_IP
+
+    def __init__(self, netns_name: str, name: str, route: Dict) -> None:
+        self.name = name
+        self.netns_name = netns_name
+        self.route = route
+
+    def __validate_routes(self, route_test: Dict) -> int:
+        """Check to make sure all elements exist that are needed for a valid route, then validate those elements"""
+        if (route_test["dest_prefix"]) and (
+            route_test["next_hop_ip"] or route_test["exit_if_name"]
+        ):
+            # Check for bad destination prefix, if so skip route installation
+            try:
+                ip_network(route_test["dest_prefix"])
+            except ValueError:
+                LOG.error(
+                    f"{route_test['dest_prefix']} is not a valid network address, please check config"
+                )  # convert to error log
+                return 1
+            # Check if a next hop IP if specified, if so check for bad next hop address and skip route installation
+        if route_test["next_hop_ip"]:
+            try:
+                ip_address(route_test["next_hop_ip"])
+            except ValueError:
+                LOG.error(
+                    f"{route_test['next_hop_ip']} is not a valid ip address, please check config"
+                )  # convert to error log
+                return 1
+        return 0
+
+    def __route_exists(self, route_prefix: str) -> bool:
+        """Checks if route exists already (maintain idempotency)"""
+        route_cmd = [
+            self.IP,
+            "netns",
+            "exec",
+            self.netns_name,
+            self.IP,
+            "route",
+            "show",
+            route_prefix,
+        ]
+        cp = (check_output(route_cmd)).decode("utf-8")
+        if cp == "":
+            return 0
+        else:
+            LOG.info(f"{route_prefix} already exists in route table, skipping...")
+            return 1
+
+    def get_route(self) -> List:
+        """Generate cmd list for use with ns class"""
+        if self.__validate_routes(self.route) == 0:
```

Comment:
Pro tip - Always try to do the inverse here with if statements
- It's generally preferred to check for error so normal good logical code has as much horizontal space as possible in our ~88 chars.
```suggestion
        if self.__validate_routes(self.route):  # i.e. we're truthy - thus an error
           LOG.error("Shit not valid")
           return []
         
         if self.route_exists()
           LOG.error("Route already there bra")
           return []
         ...
        
```

Reply:
Let me know what you think of the changes I made here.  Tried to improve the logic and readability.